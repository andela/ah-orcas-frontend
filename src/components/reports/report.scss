.report {
  margin: 70px auto;
  max-width: 1126px;
  padding: 70px 0;
}

.report-button {
  background: rgb(46, 186, 174);
  border-radius: 17px;
  float: right;
  font-size: 2rem;
  margin-right: 26px;
  margin-top: 1%;
  width: 20%;
}
