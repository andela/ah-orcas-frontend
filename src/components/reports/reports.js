/* eslint-disable react/destructuring-assignment */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Form, FormGroup, Input } from 'reactstrap';
import './report.scss';
import { reportActions } from '../../actions/report.actions';

export class Reports extends Component {
  constructor(props) {
    super(props);
    this.state = {
      title: '',
      report: '',
    };
    this.handleChange = this.handleChange.bind(this);
  }

  handleChange = (event) => {
    this.setState({
      [event.target.id]: event.target.value,
    });
  };

  handleSubmit = (event) => {
    event.preventDefault();
    // eslint-disable-next-line prefer-destructuring
    const title = this.state.title;
    const body = this.state.report;
    const data = {
      title,
      body,
    };
    reportActions(data);
    this.setState({ title: '', report: '' });
  };

  render() {
    return (
      <div className="report">
        <Form onSubmit={this.handleSubmit}>
          <FormGroup>
            <Input
              type="text"
              name="title"
              id="title"
              className="form-control description"
              placeholder="Enter Subject"
              required
              onChange={this.handleChange}
            />
          </FormGroup>
          <FormGroup>
            <Input
              type="textarea"
              rows="18"
              cols="50"
              name="text"
              id="report"
              className="form-control description"
              placeholder="Enter your report"
              required
              onChange={this.handleChange}
            />
          </FormGroup>

          <button type="submit" className="btn report-button ">
           Submit
          </button>
        </Form>
      </div>
    );
  }
}

export default connect(null)(Reports);
