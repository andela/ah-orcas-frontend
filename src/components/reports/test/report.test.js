import React from 'react';
import { shallow } from 'enzyme';
import { Reports } from '../reports';

function setup() {
  const props = {
    payload: jest.fn(),
    dispatch: jest.fn(),
  };
  const enzymeWrapper = shallow(<Reports {...props} />);
  return {
    enzymeWrapper,
  };
}
describe('snapshot test for reports', () => {
  describe('snapshot test', () => {
    it('should create a snapshot test for reports', () => {
      const { enzymeWrapper } = setup();

      enzymeWrapper.find('#report').simulate('change', {
        target: { id: 'report', value: 'something' },
      });

      expect(enzymeWrapper.state().report).toEqual('something');
      expect(enzymeWrapper).toMatchSnapshot();
    });

    it('should submit the report data', () => {
      const { enzymeWrapper } = setup();

      enzymeWrapper.find('#report').simulate('change', {
        target: { id: 'report', value: 'something' },
      });

      enzymeWrapper.find('#title').simulate('change', {
        target: { id: 'title', value: 'some title' },
      });

      enzymeWrapper.find('.report-button').simulate('click');

      setTimeout(() => {
        expect(enzymeWrapper.state().report).toEqual('');
      }, 1000);
    });
  });
});
