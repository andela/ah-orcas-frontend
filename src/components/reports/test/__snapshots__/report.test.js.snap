// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test for reports snapshot test should create a snapshot test for reports 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Reports
    dispatch={[MockFunction]}
    payload={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Form
        onSubmit={[Function]}
        tag="form"
      >
        <FormGroup
          tag="div"
        >
          <Input
            className="form-control description"
            id="title"
            name="title"
            onChange={[Function]}
            placeholder="Enter Subject"
            required={true}
            type="text"
          />
        </FormGroup>
        <FormGroup
          tag="div"
        >
          <Input
            className="form-control description"
            cols="50"
            id="report"
            name="text"
            onChange={[Function]}
            placeholder="Enter your report"
            required={true}
            rows="18"
            type="textarea"
          />
        </FormGroup>
        <button
          className="btn report-button "
          type="submit"
        >
          Submit
        </button>
      </Form>,
      "className": "report",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <FormGroup
            tag="div"
          >
            <Input
              className="form-control description"
              id="title"
              name="title"
              onChange={[Function]}
              placeholder="Enter Subject"
              required={true}
              type="text"
            />
          </FormGroup>,
          <FormGroup
            tag="div"
          >
            <Input
              className="form-control description"
              cols="50"
              id="report"
              name="text"
              onChange={[Function]}
              placeholder="Enter your report"
              required={true}
              rows="18"
              type="textarea"
            />
          </FormGroup>,
          <button
            className="btn report-button "
            type="submit"
          >
            Submit
          </button>,
        ],
        "onSubmit": [Function],
        "tag": "form",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Input
              className="form-control description"
              id="title"
              name="title"
              onChange={[Function]}
              placeholder="Enter Subject"
              required={true}
              type="text"
            />,
            "tag": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "form-control description",
              "id": "title",
              "name": "title",
              "onChange": [Function],
              "placeholder": "Enter Subject",
              "required": true,
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Input
              className="form-control description"
              cols="50"
              id="report"
              name="text"
              onChange={[Function]}
              placeholder="Enter your report"
              required={true}
              rows="18"
              type="textarea"
            />,
            "tag": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "form-control description",
              "cols": "50",
              "id": "report",
              "name": "text",
              "onChange": [Function],
              "placeholder": "Enter your report",
              "required": true,
              "rows": "18",
              "type": "textarea",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Submit",
            "className": "btn report-button ",
            "type": "submit",
          },
          "ref": null,
          "rendered": "Submit",
          "type": "button",
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Form
          onSubmit={[Function]}
          tag="form"
        >
          <FormGroup
            tag="div"
          >
            <Input
              className="form-control description"
              id="title"
              name="title"
              onChange={[Function]}
              placeholder="Enter Subject"
              required={true}
              type="text"
            />
          </FormGroup>
          <FormGroup
            tag="div"
          >
            <Input
              className="form-control description"
              cols="50"
              id="report"
              name="text"
              onChange={[Function]}
              placeholder="Enter your report"
              required={true}
              rows="18"
              type="textarea"
            />
          </FormGroup>
          <button
            className="btn report-button "
            type="submit"
          >
            Submit
          </button>
        </Form>,
        "className": "report",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <FormGroup
              tag="div"
            >
              <Input
                className="form-control description"
                id="title"
                name="title"
                onChange={[Function]}
                placeholder="Enter Subject"
                required={true}
                type="text"
              />
            </FormGroup>,
            <FormGroup
              tag="div"
            >
              <Input
                className="form-control description"
                cols="50"
                id="report"
                name="text"
                onChange={[Function]}
                placeholder="Enter your report"
                required={true}
                rows="18"
                type="textarea"
              />
            </FormGroup>,
            <button
              className="btn report-button "
              type="submit"
            >
              Submit
            </button>,
          ],
          "onSubmit": [Function],
          "tag": "form",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Input
                className="form-control description"
                id="title"
                name="title"
                onChange={[Function]}
                placeholder="Enter Subject"
                required={true}
                type="text"
              />,
              "tag": "div",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "form-control description",
                "id": "title",
                "name": "title",
                "onChange": [Function],
                "placeholder": "Enter Subject",
                "required": true,
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Input
                className="form-control description"
                cols="50"
                id="report"
                name="text"
                onChange={[Function]}
                placeholder="Enter your report"
                required={true}
                rows="18"
                type="textarea"
              />,
              "tag": "div",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "form-control description",
                "cols": "50",
                "id": "report",
                "name": "text",
                "onChange": [Function],
                "placeholder": "Enter your report",
                "required": true,
                "rows": "18",
                "type": "textarea",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Submit",
              "className": "btn report-button ",
              "type": "submit",
            },
            "ref": null,
            "rendered": "Submit",
            "type": "button",
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
